runtime/doc/dev_theme.txt

```diff
@@ -0,0 +1,96 @@
+*dev_theme.txt*          Nvim
+
+
+                            NVIM REFERENCE MANUAL
+
+
+Nvim theme style guide                                        *dev-theme*
+
+This is style guide for developers working on Nvim's default color scheme.
+
+License: CC-By 3.0 https://creativecommons.org/licenses/by/3.0/
+
+                                      Type |gO| to see the table of contents.
+
+==============================================================================
+Design
+
+Be "Neovim branded", i.e. have mostly "green-blue" feel plus one/two colors
+reserved for very occasional user attention.
+
+Be extra minimal for 'notermguicolors' (256 colors) while allowing a bit more
+shades for when 'termguicolors' is set (true colors).
+
+Be accessible, i.e. have enough contrast ratio. This means to have at least 7
+for |hl-Normal| and 4.5 for some common cases (|hl-Visual|, `Comment` with set
+'cursorline', colored syntax, `Diff*`, |hl-Search|).
+
+Be dark and light variants a simple exchange of dark and light palettes.
+
+Be usable, i.e. provide enough visual feedback for common objects.
+
+
+==============================================================================
+Palettes
+
+There are two separate palettes: dark and light. They all contain same set of
+colors exported as `NvimDark*` and `NvimLight*` colors respectively.
+
+Dark/light palette is used for background in dark/light color scheme and for
+foreground in light/dark color scheme. This introduces recognizable visual
+system without too standing out.
+
+Actual computation of palettes should be done in a perceptually uniform color
+space. Oklch is a good choice.
+
+Each palette has the following colors (descriptions are for dark background;
+reverse for light one):
+
+- Four shades of "colored" greys for general UI. In 256 colors they are exact
+  greys, with true colors - shades of "cool" grey.
+    - Dark ones (from darkest to lightest) are reserved as background for
+      |hl-NormalFloat| (considered as "black"), |hl-Normal| (background),
+      |hl-CursorLine|, |hl-Visual|.
+    - Light ones (also from darkest to lightest) are reserved for
+      `Comment`, |hl-StatusLine|/|hl-TabLine|, |hl-Normal| (foreground),
+      and color considered as "white".
+
+- Six colors to be enough to comply with terminal colors: red, yellow, green,
+  cyan, blue, magenta.
+  They should have (reasonably) the same lightness and chroma to make them
+  visually coherent. Lightness should be equal to palette's basic grey (which
+  is used for |hl-Normal|). They should have (reasonably) different hues to
+  make them visually separable.
+
+- Each palette color should have the 256 colors variant with closest color
+  computed based on the perceptually uniform distance measure.
+
+
+==============================================================================
+Highlight groups
+
```

Comment:
Here (as in the design section), you could structure this as 

```
Use

* 8 shades of grey ...

* bold text ...

```

This makes the keywords stand out more.